packages/note/src/react/use-hover.js

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useRef, useState } from 'react'
+
+export default function useHover() {
+  const [isHovered, setIsHovered] = useState(false)
+  const ref = useRef(null)
+
+  const handleMouseOver = () => setIsHovered(true)
+  const handleMouseOut = () => setIsHovered(false)
+
+  useEffect(
+    function setup() {
+      const node = ref.current
+
+      if (!node) return
+
+      node.addEventListener('mouseover', handleMouseOver)
```

Comment:
This is an interesting approach.  Seems like the more react approach would be to avoid refs entirely (since they're just an escape hatch to the dom) and instead use normal event handlers as props (ie, `onMouseOver`, etc). 

What do you think?

Reply:
yeah that makes sense. this hook could be shared but that seems overly complicated and not actually what we need. i'm in a hook all the things mode...i need to step back and think sometimes 😉